cmd/snap-update-ns/utils.go

```diff
@@ -110,7 +110,7 @@ func secureMkDir(fd int, segments []string, i int, perm os.FileMode, uid, gid in
 
 	const openFlags = syscall.O_NOFOLLOW | syscall.O_CLOEXEC | syscall.O_DIRECTORY
 
-	if err = sysMkdirat(fd, segment, uint32(perm)); err != nil {
+	if err = sysMkdirat(fd, segment, uint32(perm.Perm())); err != nil {
```

Comment:
No, I just noticed after opening the other PR and I somehow landed the change here. Technically we want the Perm part of the mode so this is just for correctness in case of abuse.